### DISCUSSION TITLE
How can I access and store my custom profile in my MongoDB?

### DISCUSSION THREAD

**Discussion by rajbirjohar** [nextauthjs/next-auth#2384] [Help]
Hey guys, I've pretty much read through all the documentation and looked at the provided example but I cannot figure out/find out how to actually store my custom user profile (using the Github provider) in my MongoDB database because no matter what, it ignores my custom user profile. But I was able to successfully read it from the Github provider. 

 I just want to be able to add the `profile.login`/`user login` (Github username) to my model. I also am lost on how to access it via my session such as `session.user.login`. 

Below is the necessary code:

```js
// [...nextauth.js]
import NextAuth from 'next-auth'
import Providers from 'next-auth/providers'
import { Profiler } from 'react'

export default NextAuth({
  // Configure one or more authentication providers
  providers: [
    Providers.GitHub({
      clientId: process.env.GITHUB_ID,
      clientSecret: process.env.GITHUB_SECRET,
      scope: 'read:user',
      profile(profile) {
        console.log(profile.login)
        return {
          id: profile.id,
          name: profile.name || profile.login,
          login: profile.login,
          email: profile.email,
          image: profile.avatar_url,
        }
      },
    }),
    // ...add more providers here
  ],
  // A database is optional, but required to persist accounts in a database
  database: process.env.MONGODB_URI,
})
```

I'm thinking I have to make use of the callback functions but I'll be honest, the documentation on it is oblivious to me and I have not been able to find a super simple working example of using them. 

I'm mainly confused on if I should/could use jwt even if i have a database that persists users. Then how can use the session callback to access the user items and if I would also be storing them in the DB here or if that involves the signIn callback.

I would much appreciate it if any of you could show me some example code since I find that helps me a lot. Thanks!

**Comment by balazsorban44** [comment]
https://github.com/nextauthjs/adapters/issues/151 is probably related.

Although adding `login` shouldn't be necessary in your case, since it is already used by default:

https://github.com/nextauthjs/next-auth/blob/main/src/providers/github.js

**Reply by rajbirjohar** [reply]
Yeah I studied the default profile extensively and well its not really used by default because Github specifies is either as the github name (real name) or the github login (username) if the name hasn't been provided (i.e. new accounts). I'm trying to force it to record the username since atleast that's something all accounts have. I would also like to do email but apparently it's another set of challenges. Do you have any proposal for this?

**Reply by rajbirjohar** [reply]
Ah, okay now I see what you mean by default. I can remove the `profile.name ||` operand to just give me `profile.login`. This will suffice but is there any way to possibly add extra fields? Or referencing the issue you mentioned, will be reserved for v4?

Edit: Also, when trying to explicitly request the email using the given model above, there are some instances where the email also won't save to the db. This is because Github users can hide their email correct? Do you have any suggestions on how I can go about that?
